src/lib.rs

```diff
@@ -118,22 +118,64 @@ pub fn lock_contended_error() -> Error {
     sys::lock_error()
 }
 
+/// FsStats contains some common stats about a file system.
+pub struct FsStats {
+    free_space: u64,
+    available_space: u64,
+    total_space: u64,
+    allocation_granularity: u64,
+}
+
+impl FsStats {
+    /// Get the stats of the file system containing the provided path.
+    pub fn statvfs<P>(path: P) -> Result<FsStats> where P: AsRef<Path> {
```

Comment:
Could you make this a free function so that it more closely matches the rest of the API?